.github/workflows/scripts/generate-weekly-report.js

```diff
@@ -0,0 +1,428 @@
+  "componentData": ${JSON.stringify(componentData, null, 2)}
+}
+</pre>
+</details>`);
+  const report = out.join('\n');
+  return report;
+}
+
+async function createIssue({ github, lookbackData, report, context }) {
+  const title = `Weekly Report: ${lookbackData.sevenDaysAgo.toISOString().slice(0, 10)} - ${lookbackData.midnightYesterday.toISOString().slice(0, 10)}`;
+  return github.rest.issues.create({
+    owner: context.payload.repository.owner.login,
+    repo: "opentelemetry-collector-contrib",
+    title,
+    body: report,
+    labels: ["report"]
+  })
+}
+
+async function getLastWeeksReport({ github, since, context }) {
+  const issues = await github.rest.issues.listForRepo({
+    owner: context.payload.repository.owner.login,
+    repo: 'opentelemetry-collector-contrib',
+    state: 'all', // To get both open and closed issues
+    labels: ["report"],
+    since: since.toISOString(),
+    per_page: 1,
+    direction: "asc"
+  });
+  if (issues.data.length === 0) {
+    return null;
+  } 
+  // grab earliest issue if multiple
+  return issues.data[0];
+}
+
+function parseJsonFromText(text) {
+  // Use regex to find the JSON data within the <pre></pre> tags
+  const regex = /<pre>\s*([\s\S]*?)\s*<\/pre>/;
+  const match = text.match(regex);
+
+  if (match && match[1]) {
+    // Parse the found string to JSON
+    return JSON.parse(match[1]);
+  } else {
+    throw new Error("JSON data not found");
+  }
+}
+
+async function processIssues({ github, context, lookbackData }) {
+  const issuesData = await getIssuesData({github, context});
+
+  const prevReportLookback = new Date(lookbackData.sevenDaysAgo)
+  prevReportLookback.setDate(prevReportLookback.getDate() - 7)
+  const previousReportIssue = await getLastWeeksReport({github, since: prevReportLookback, context});
+  // initialize to zeros
+  let previousReport = null;
+
+  if (previousReportIssue !== null) {
+    const {created_at, id, url, title} = previousReportIssue;
+    debug({ msg: "previous issue", created_at, id, url, title })
+    previousReport = parseJsonFromText(previousReportIssue.body)
+  }
+
+  return {issuesData, previousReport}
+
+
+}
+
+const findFilesByName = (startPath, filter) => {
+  let results = [];
+
+  // Check if directory exists
+  let files;
+  try {
+      files = fs.readdirSync(startPath);
+  } catch (error) {
+      console.error("Error reading directory: ", startPath, error);
+      return [];
+  }
+
+  for (let i = 0; i < files.length; i++) {
+      const filename = path.join(startPath, files[i]);
+      let stat;
+      try {
+          stat = fs.lstatSync(filename);
+      } catch (error) {
+          console.error("Error stating file: ", filename, error);
+          continue;
+      }
+      
+      if (stat.isDirectory()) {
+          const innerResults = findFilesByName(filename, filter); // Recursive call
+          results = results.concat(innerResults);
+      } else if (filename.indexOf(filter) >= 0) {
```

Comment:
Why only check that "metadata.yaml" is part of the file name, as opposed to check that the file name equals "metadata.yaml"?
```suggestion
      } else if (path.basename(filename) == filter) {
```

Reply:
good point. updated